math/mathcore/inc/TMath.h

```diff
@@ -1002,6 +1006,50 @@ Iterator TMath::LocMin(Iterator first, Iterator last) {
    return std::min_element(first, last);
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Returns the one-dimensional gradient of a vector with length n. The default step size h is 1.
+
+template <typename T>
+T *TMath::Gradient(Long64_t n, T *f, T h)
+{
+   Long64_t i = 1;
+   T *result = new T[n + 1];
+
+   // Forward difference
+   result[0] = (f[1] - f[0]) / h;
+
+   // Central difference
+   while (i < n - 1) {
+      result[i] = (f[i + 1] - f[i - 1]) / (2 * h);
+      i++;
+   }
+   // Backward difference
+   result[i] = (f[i] - f[i - 1]) / h;
+   return result;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Returns the Laplacian of a vector with length n. The default step size h is 1.
+
+template <typename T>
+T *TMath::Laplacian(Long64_t n, T *f, T h)
+{
+   int i = 1;
+   T *result = new T[n + 1];
```

Comment:
```suggestion
   T *result = new T[n - 1];
```

Reply:
Same as above.

Reply:
Edited. Add also here the same documentation as for Gradient thanks!!